deflate.h

```diff
@@ -334,11 +334,11 @@ extern const uint8_t ZLIB_INTERNAL _dist_code[];
 #define likely(x)       (x)
 #define unlikely(x)     (x)
 
-int __inline __builtin_ctzl(unsigned long mask)
+uint64_t __inline __builtin_ctzll(uint64_t mask)
```

Comment:
Nit: Since index is an unsigned long and at most 64, can you replace the return value here with unsigned long and remove the cast to int in the return expression? I doubt it matters for codegen but it should look slightly cleaner.